src/coordination/coordinator_instance.cpp

```diff
@@ -214,78 +223,148 @@ auto CoordinatorInstance::ShowInstances() const -> std::vector<InstanceStatus> {
+    // address so for them we can just retrieve their socket address.
+    if (instance.management_server.GetResolvedIPAddress() == "0.0.0.0") {
+      return fmt::format("{}:{}", instance.coordinator_hostname, instance.management_server.GetPort());
+    }
+    return instance.management_server.SocketAddress();
+  };
+
   auto const coord_instance_to_status =
-      [this, &stringify_coord_health, &get_coord_role, &get_coordinator_server,
-       &get_bolt_server](CoordinatorToCoordinatorConfig const &instance) -> InstanceStatus {
+      [this, &stringify_coord_health, &get_coord_role, &get_coordinator_server, &get_bolt_server,
+       &get_management_server](CoordinatorToCoordinatorConfig const &instance) -> InstanceStatus {
     auto const curr_leader = raft_state_->GetLeaderId();
-    return {.instance_name = fmt::format("coordinator_{}", instance.coordinator_id),
-            .coordinator_server = get_coordinator_server(instance),  // show non-resolved IP
-            .bolt_server = get_bolt_server(instance),                // show non-resolved IP
-            .cluster_role = get_coord_role(instance.coordinator_id, curr_leader),
-            .health = stringify_coord_health(instance),
-            .last_succ_resp_ms = raft_state_->CoordLastSuccRespMs(instance.coordinator_id).count()};
+    return {
+        .instance_name = fmt::format("coordinator_{}", instance.coordinator_id),
+        .coordinator_server = get_coordinator_server(instance),  // show non-resolved IP
+        .management_server = get_management_server(instance),    // show non-resolved IP
+        .bolt_server = get_bolt_server(instance),                // show non-resolved IP
+        .cluster_role = get_coord_role(instance.coordinator_id, curr_leader),
+        .health = stringify_coord_health(instance),
+        .last_succ_resp_ms = raft_state_->CoordLastSuccRespMs(instance.coordinator_id).count(),
+
+    };
   };
 
-  auto instances_status = utils::fmap(raft_state_->GetCoordinatorInstances(), coord_instance_to_status);
+  return utils::fmap(raft_state_->GetCoordinatorInstances(), coord_instance_to_status);
+}
 
-  if (raft_state_->IsLeader()) {
-    auto const stringify_repl_role = [this](ReplicationInstanceConnector const &instance) -> std::string {
-      if (!instance.IsAlive()) return "unknown";
-      if (raft_state_->IsCurrentMain(instance.InstanceName())) return "main";
-      return "replica";
-    };
+auto CoordinatorInstance::GetAllInstancesStatusAsFollower() const -> std::vector<InstanceStatus> {
+  auto instances_status = GetCoordinatorsInstanceStatus();
+  spdlog::trace("Processed all coordinators, processing replication instances as follower.");
+  auto const stringify_inst_status = [raft_state_ptr =
+                                          raft_state_.get()](ReplicationInstanceState const &instance) -> std::string {
+    if (raft_state_ptr->IsCurrentMain(instance.config.instance_name)) {
```

Comment:
just add a comment here, instance is probably down that's why we return unknown. My mistake it wasn't added before